modules/substrate/src/lib.rs

```diff
@@ -241,6 +303,16 @@ pub trait BridgeStorage {
 	/// Check if a particular header is known to the pallet.
 	fn header_exists(&self, hash: <Self::Header as HeaderT>::Hash) -> bool;
 
+	/// Return a header which requires a justification. A header will require
+	/// a justification when it enacts an new authority set.
+	fn unfinalized_header(&self) -> Option<<Self::Header as HeaderT>::Hash>;
+
+	/// Mark a header as eventually requiring a justification.
+	fn update_unfinalized_header(&mut self, hash: <Self::Header as HeaderT>::Hash);
+
+	/// Mark that we have received a justification for a header which required one.
+	fn clear_unfinalized_header(&mut self);
```

Comment:
Why not `Option` in the `update_*` API?

Reply:
I thought it might be a little confusing, but after trying it out I think it's fine